lib/gli/command_support.rb

```diff
@@ -166,14 +166,10 @@ def get_action(arguments)
 
     def generate_error_action(arguments)
       lambda { |global_options,options,arguments|
-        if am_subcommand?
-          if arguments.size > 0
-            raise UnknownCommand,"Unknown command '#{arguments[0]}'"
-          else
-            raise BadCommandLine,"Command '#{name}' requires a subcommand"
-          end
-        elsif have_subcommands?
+        if have_subcommands?
```

Comment:
I'm pretty sure that reordering these conditionals is safe but please double-check that!

Reply:
It seems to be breaking this test:

https://github.com/davetron5000/gli/blob/main/test/tc_subcommands.rb#L149

(sorry about the weird test framework—Given/When/Then don't do anything so the test is vanilla Test::Unit really)

That test is wanting to see that if the command has a subcommand, but extra args were given, the message is "Unknown command".  I guess that error check has precedence, so I think this `if` statement does need to be kinda (annoyingly) complex.

Reply:
Ah whoops! I think I can just reorder the `if` and `elsif` parts of the conditional—let me know if that seems right now.

(I'm trying out GitHub's "make a code change and PR right from the browser" feature but a downside is I can't run tests. Let me know if this process is too annoying for you.)